src/particles.jl

```diff
@@ -306,10 +304,22 @@ for PT in (:Particles, :StaticParticles, :WeightedParticles)
         Base.:\(p::Vector{<:$PT}, p2::Vector{<:$PT}) = Matrix(p)\Matrix(p2) # Must be here to be most specific
     end
 
+    @eval Base.promote_rule(::Type{S}, ::Type{$PT{T,N}}) where {S<:Number,T,N} = $PT{promote_type(S,T),N}
+    @eval Base.promote_rule(::Type{Bool}, ::Type{$PT{T,N}}) where {T,N} = $PT{promote_type(Bool,T),N}
 
-
+    for PT2 in (:Particles, :StaticParticles, :WeightedParticles)
```

Comment:
```suggestion
    for PT2 in (:Particles, :StaticParticles)
```
I just pulled the PR that removes the weighted particles